File: src/test/resources/oracle/commits/rm-action-test-0ce4051977047c6311f397055a07218a7318d87c/python/order_processor.py
class OrderProcessor:
    def __init__(self):
        self._tax_rate = 0.08

    def build_receipt(self, customer_name, unit_price, quantity, discount):
        subtotal = unit_price * quantity
        discounted_subtotal = subtotal - discount
        tax = int(discounted_subtotal * self._tax_rate)
        total = discounted_subtotal + tax

        header = self._format_header(customer_name)
        dollars = self._cents_to_dollars(total)
        status = self._format_payment_status(total, discount)

        return (
            f"{header}\n"
            f"Subtotal: {subtotal}\n"
            f"Discount: {discount}\n"
            f"Tax: {tax}\n"
            f"Total: ${dollars}\n"
            f"{status}"
        )

    def _format_header(self, customer_name):
        return "Receipt for " + customer_name.strip().upper()

    def _cents_to_dollars(self, cents):
        return cents / 100.0

    def _format_payment_status(self, total, discount):
        if total > 0 and discount > 0:
            return "Discount applied"
        if total > 0:
            return "Payment required"
        return "No payment required"

File: src/test/resources/oracle/commits/rm-action-test-9e5e9f27809d45b6630ea0e8bbe409a1649c77ea/javascript/OrderProcessor.js
import { PricingPolicy } from "./PricingPolicy.js";
import { ReceiptFormatter } from "./ReceiptFormatter.js";

export class OrderProcessor {
    constructor() {
        this._formatter = new ReceiptFormatter();
    }

    buildReceipt(buyerName, unitPrice, quantity, discount) {
        const total = this.calculateTotal(unitPrice, quantity, discount);
        const subtotal = unitPrice * quantity;
        const discountedSubtotal = subtotal - discount;
        const tax = Math.floor(discountedSubtotal * new PricingPolicy().taxRate);

        const header = this._formatter.createReceiptHeader(buyerName);
        const dollars = total / 100.0;
        const status = this.describePaymentStatus(total, discount);

        return (
            `${header}\n` +
            `Subtotal: ${subtotal}\n` +
            `Discount: ${discount}\n` +
            `Tax: ${tax}\n` +
            `Total: $${dollars}\n` +
            `${status}`
        );
    }

    calculateTotal(unitPrice, quantity, discount) {
        const subtotal = unitPrice * quantity;
        const discountedSubtotal = subtotal - discount;
        const tax = Math.floor(discountedSubtotal * new PricingPolicy().taxRate);
        return discountedSubtotal + tax;
    }

    describePaymentStatus(total, discount) {
        if (total > 0 && discount > 0) {
            return "Discount applied";
        }
        if (total > 0) {
            return "Payment required";
        }
        return "No payment required";
    }
}
